src/impls/impl-posix/sys.cpp

```diff
@@ -442,6 +442,21 @@ Res<> exit(i32 res) {
     return Ok();
 }
 
+Res<Mime::Url> pwd() {
+    usize const bufSize = 512;
+    Buf<char> buf(bufSize);
+
+    if (::getcwd(buf.buf(), bufSize) == NULL)
```

Comment:
Can you take a look in the informative part of this spec for the properway to invoke this function:

https://pubs.opengroup.org/onlinepubs/9799919799/functions/getcwd.html#:~:text=The%20following%20sections%20are%20informative.

Reply:
top thx again! super resource!

however, if we have to call this method every time (instead of caching it as suggested in the PR) and, every time we call it, we allocate the PATH_MAX value (if its set), do you think this could hurt performance?

Reply:
Correctness goes before performance